Remove custom palettes from building placement previews.

pchote opened the pull request:
Building on #18550, this PR replaces the custom palettes in PBOG previews with an explicit alpha value to make them compatible with with 32bit sprites.

The testcase commit demonstrates that target flashes and WithColoredOverlay (from #18550) and building placement previews work as expected for the D2k R16 sprites (loaded as 32bit). You will need to provide your own R16 files from an original game install.

pchote commented:
Rebased and ready for review.

I have moved the testcase to https://github.com/pchote/OpenRA/tree/d2k-r16  where it can benefit from the mixed RGBA/Indexed terrain support from #18982 and also demonstrate #19012.

teinarss commented on OpenRA.Mods.Common/Traits/Buildings/FootprintPlaceBuildingPreview.cs:
```diff
@@ -9,6 +9,7 @@
  */
 #endregion
 
+using System;
```
![image](https://user-images.githubusercontent.com/19173272/104517002-992ace80-55f5-11eb-9b5a-28434c7acf0d.png)

pchote commented:
Fixed.

abcdefg30 commented on OpenRA.Mods.Common/Traits/Buildings/ActorPreviewPlaceBuildingPreview.cs:
```diff
@@ -25,12 +25,8 @@ public class ActorPreviewPlaceBuildingPreviewInfo : FootprintPlaceBuildingPrevie
-		[Desc("Custom palette is a player palette BaseName.")]
-		public readonly bool OverridePaletteIsPlayerPalette = true;
+		[Desc("Custom opacity to apply to the actor preview.")]
+		public readonly float PreviewAlpha = 1.0f;
```
We use `1f` everywhere else, why the `1.0` suddenly?

abcdefg30 reviewed (commented):
Imo it would be good if the update rule mentioned the locations it removed nodes at (as we usually do) since it does not automatically replace the values.

pchote committed:
Remove custom palettes from building placement previews.

pchote commented:
Rebased and updated.

New update messages:
```
RemovePlaceBuildingPalette: *PlaceBuildingPreview palette overrides have been removed.
   Updating mod... COMPLETE
   Updating system maps... COMPLETE
   Manual changes are required to complete this update:
    * The *Palette fields have been removed from the *PlaceBuildingPreview traits.
      You may wish to inspect the following definitions and define new Alpha or
      LineBuildSegmentAlpha properties as appropriate to recreate transparency effects:
       * ^Building (rules/defaults.yaml)
       * ^Wall (rules/defaults.yaml)
       * WEAP (rules/structures.yaml)
       * SAM (rules/structures.yaml)
```

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a7116c1232682fea806f31508398f5360d16130a...4023137d33af88fe810704d8fb0c1863f9204b34
